prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -442,54 +443,50 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
     suspend fun postReport(
         message: ByteArray,
         fileName: String,
-        restUrl: String,
+        restTransportInfo: RESTTransportType,
         headers: Map<String, String>,
         logger: Logger,
-        httpClient: HttpClient
+        httpClient: HttpClient,
+        receiver: Receiver
     ): HttpResponse {
         logger.info(fileName)
-        val boundary = "WebAppBoundary"
         httpClient.use { client ->
-            val theResponse: HttpResponse = client.post(restUrl) {
+            val theResponse: HttpResponse = client.post(restTransportInfo.reportUrl) {
                 logger.info("posting report to rest API")
                 expectSuccess = true // throw an exception if not successful
                 postHeaders(
                     headers
                 )
                 setBody(
-                    when (restUrl.substringAfterLast('/')) {
-                        // OK or NBS
-                        "hl7", "reports" -> {
-                            TextContent(message.toString(Charsets.UTF_8), ContentType.Text.Plain)
-                        }
-                        // Flexion
-                        "demographics" -> {
-                            TextContent(message.toString(Charsets.UTF_8), ContentType.Application.Json)
-                        }
-                        "orders" -> {
-                            TextContent(message.toString(Charsets.UTF_8), ContentType.Application.Json)
-                        }
-                        // WA
-                        "elr" -> {
+                    // most should use the else branch unless it's not possible on the reciever's end
+                    when (receiver.organizationName) {
+                        "wa-phd" -> {
```

Comment:
I don't like the idea of hard coding org names into our logic.

Reply:
I agree that it's not great. It was the best I could come up with for NY and WA's very specific custom logic required. In essence we are already doing the same thing but with URLs instead.

Maybe it's possible to configure some kind of function in the configuration? Not sure off the top of my head.